src/main/java/fr/xephi/authme/service/bungeecord/MessageType.java

```diff
@@ -1,19 +1,51 @@
 package fr.xephi.authme.service.bungeecord;
 
-public final class MessageType {
+public enum MessageType {
+    // Placeholders
+    UNKNOWN("unknown", false),
+    // Broadcast messages
+    REFRESH_PASSWORD("refresh.password", true),
+    REFRESH_SESSION("refresh.session", true),
+    REFRESH_QUITLOC("refresh.quitloc", true),
+    REFRESH_EMAIL("refresh.email", true),
+    REFRESH("refresh", true),
+    REGISTER("register", true),
+    UNREGISTER("unregister", true),
+    // Bungee-only outgoing messages
+    // TODO #1614: should be broadcasts, AuthMeBungee needs to be adapted.
+    LOGIN("login", false),
+    LOGOUT("logout", false),
+    // Bungee-only incoming messages
+    BUNGEE_LOGIN("bungeelogin", false);
 
-    public static final String LOGIN = "login";
-    public static final String LOGOUT = "logout";
-    public static final String REGISTER = "register";
-    public static final String UNREGISTER = "unregister";
-    public static final String REFRESH_PASSWORD = "refresh.password";
-    public static final String REFRESH_SESSION = "refresh.session";
-    public static final String REFRESH_QUITLOC = "refresh.quitloc";
-    public static final String REFRESH_EMAIL = "refresh.email";
-    public static final String REFRESH = "refresh";
-    public static final String BUNGEE_LOGIN = "bungeelogin";
+    private final String id;
+    private final boolean broadcast;
 
-    private MessageType() {
+    MessageType(String id, boolean broadcast) {
+        this.id = id;
+        this.broadcast = broadcast;
     }
 
+    public String getId() {
+        return id;
+    }
+
+    public boolean isBroadcast() {
+        return broadcast;
+    }
+
+    /**
+     * Returns the MessageType with the given ID.
+     *
+     * @param id the message type id.
+     * @return the MessageType with the given id, MessageType.UNKNOWN if invalid.
+     */
+    public static MessageType fromId(String id) {
+        for(MessageType current : values()) {
+            if(current.getId().equals(id)) {
+                return current;
+            }
+        }
+        return UNKNOWN;
```

Comment:
Discussion: does it make sense to return an enum entry here that I guess only exists for this method? Other alternatives could be to return `null` here, or an Optional

Reply:
not a big fun of nulls, and an optional is a little overkill in this case, idk

Reply:
I saw you changed it – thank you – the reason it irked me is that it‘s an additional entry in the enum but it‘s not really a message type; it‘s a marker for an invalid state, so to speak.